OIPA/iati_organisation/parser/organisation_2_03.py

```diff
@@ -942,6 +942,147 @@ def iati_organisations__iati_organisation__recipient_region_budget__budget_line_
+        period_end_date = self.validate_date(period_end_date)
+        if not period_end_date:
+            raise FieldValidationError(
+                "RecipientCountryBudget",
+                "iso-date",
+                "is not in correct range.",
+                None,
+                None,
+            )
+        value_element = element.xpath("value")
+        if len(value_element) is not 1:
+            raise ParserError("RecipientCountryBudget",
+                              "value",
+                              "must occur once and only once.")
+        value = self.guess_number("RecipientCountryBudget", value_element[
+            0].text)
+
+        currency = value_element[0].attrib.get("currency")
+        if not currency:
+            currency = getattr(self.get_model("Organisation"),
+                               "default_currency")
+            if not currency:
+                raise RequiredFieldError(
+                    "RecipientCountryBudget",
+                    "currency",
+                    "must specify default-currency on iati-organisation or "
+                    "as currency on the element itself."
+                )
+        else:
+            currency = self.get_or_none(codelist_models.Currency,
+                                        code=currency)
+            if not currency:
+                raise FieldValidationError(
+                    "RecipientCountryBudget",
+                    "currency",
+                    "not found on the accompanying codelist.",
+                    None,
+                    None,
+                )
+
+        value_date = value_element[0].attrib.get("value-date")
+        if value_date is None:
+            raise RequiredFieldError("RecipientCountryBudget", "value-date",
+                                     "required field missing.")
+        value_date = self.validate_date(value_date)
+        if not value_date:
+            raise FieldValidationError(
+                "RecipientCountryBudget",
+                "value-date",
+                "not in the correct range.",
+                None,
+                None,
+            )
+        organisation = self.get_model("Organisation")
+        recipient_country_budget = RecipientCountryBudget()
+        recipient_country_budget.organisation = organisation
+        recipient_country_budget.status = status
+        recipient_country_budget.country = recipient_country_code
+        recipient_country_budget.period_start = period_start_date
+        recipient_country_budget.period_end = period_end_date
+        recipient_country_budget.value_date = value_date
+        recipient_country_budget.currency = currency
+        recipient_country_budget.value = value
+
+        self.register_model("RecipientCountryBudget", recipient_country_budget)
+        return element
+
+    def iati_organistions__iati_organisation__recipient_country_budget__recipient_country__narrative(  # NOQA: E501
```

Comment:
This method is called `iati_organistions__iati_organisation__recipient_country_budget__recipient_country__narrative`, but you're adding a narrative for `RecipientCountryBudget` element (that does not have a child Narrative element according to [the docs](http://reference.iatistandard.org/203/organisation-standard/iati-organisations/iati-organisation/recipient-country-budget/)). This is another reaon why I think that separating methods for different elements is a much better idea than squashing them in one single method with `xpath` :) See: [this comment](https://github.com/zimmerman-zimmerman/OIPA/pull/1062/files#r257707170) and [separation of concerns](https://en.wikipedia.org/wiki/Separation_of_concerns)

Reply:
Update: I think that a separate model named `BudgetRecipientCountry` should be created here (as in Activity file we have `ActivityRecipientCountry` and `TransactionRecipientCountry`)